### PR TITLE
Custom desktop font

### PR DESCRIPTION
**Pull request by HAHH9527**
#6376

### CODE REVIEW

**Comment by SomeTroglodyte** [comment]
Could that also work on Android?
Also, I'd not read the settings here, I'd pass them down from DesktopLauncher.main where the file is read anyway.

**Comment by HAHH9527** [comment]
@SomeTroglodyte 

> Could that also work on Android?

I think this setting only run with desktop.
Because desktop default fonts are easily unclear in non-English.

>Also, I'd not read the settings here, I'd pass them down from DesktopLauncher.main where the file is read anyway.

Your mean is this?
```kotlin
class NativeFontDesktop(private val size: Int) : NativeFontImplementation {
    private val font by lazy {
        val settings = UncivGame.Current.settings
        Font(settings.desktopFontFamily, Font.PLAIN, size)
    }
...
```
or this?
```kotlin
class NativeFontDesktop(private val size: Int, private val settings: GameSettings) : NativeFontImplementation {
    private val font by lazy {
        Font(settings.desktopFontFamily, Font.PLAIN, size)
    }
...
```

**Comment by HAHH9527** [comment]
And this code only run in `desktop` module.
I didn't change other modules.

So if you don't want change setting page, We can use this setting as a hidden option.
We just need to inform the player in the docs.

Or we can add an text prompt in setting page.

**Comment by HAHH9527** [comment]
@SomeTroglodyte
Look the new commit.
We can use `getDesktopAllFonts()` to show setting.
If the list is empty, the setting does not require a custom desktop font to be displayed.

File: core/src/com/unciv/ui/utils/Fonts.kt [hunk 1]
```diff
@@ -18,8 +18,14 @@ import com.unciv.models.stats.Stat
 interface NativeFontImplementation {
     fun getFontSize(): Int
     fun getCharPixmap(char: Char): Pixmap
+    open fun getDesktopAllFonts(): List<DesktopFont> = emptyList()
 }
 
+// If save in `GameSettings` need use enName.
+// If show to user need use localName.
+// If save localName in `GameSettings` may generate garbled characters by encoding.
+data class DesktopFont(val localName: String, val enName: String)
```
Comment (HAHH9527) [inline core/src/com/unciv/ui/utils/Fonts.kt:27]: @SomeTroglodyte 
`localName` is name of language used by the user. But that maybe not ASCII code.
If use this name save in setting, this value will become garbled.

`enName` is ASCII code. But users may not necessarily recognize what this is.
Like `黑体` 's `enName` is `SimHei`.
`微软雅黑` is `Microsoft YaHei`

**Review by yairm210 — APPROVED** [review]

**Review by SomeTroglodyte — APPROVED** [review]
That `open fun` in an interface (redundant) isn't mine, is it? Otherwise, of course I approve.

**Comment by SomeTroglodyte** [comment]
People are perhaps going to notice the extra delay opening OptionsPopup on >= Q, though. 4s on a slow AVD.

**Pull request merged**